config/config_test.go

```diff
@@ -85,6 +85,138 @@ func TestServiceNameFromEnv(t *testing.T) {
 	os.Unsetenv(envServiceName)
 }
 
+
+func TestConfigFromEnv(t *testing.T) {
+	os.Setenv(envServiceName, "my-service")
+	os.Setenv(envDisabled, "false")
+	os.Setenv(envRPCMetrics, "true")
+	os.Setenv(envTags, "KEY=VALUE")
+
+	cfg := &Configuration{
+		ServiceName:         "my-config-service",
+		Disabled:            true,
+		RPCMetrics:          false,
+		Tags:                []opentracing.Tag{{"KEY-CONFIG","KEY-VALUE"}},
+	}
+
+	cfg, err := cfg.FromEnv()
+	assert.NoError(t, err)
+	assert.Equal(t, "my-service", cfg.ServiceName)
+	assert.Equal(t, false, cfg.Disabled)
+	assert.Equal(t, true, cfg.RPCMetrics)
+	assert.Equal(t, "KEY", cfg.Tags[0].Key)
+	assert.Equal(t, "VALUE", cfg.Tags[0].Value)
```

Comment:
I suggest to change this test to be more comprehensive, as follows:
  1. manually initialize Config struct (as you do already)
  2. do not set any env vars, just call `cfg.FromEnv` and assert that the values remained as configured
  3. set env vars, call cfg.FromEnv again
  4. assert that values changed

Right now the test is missing step 2.

Reply:
I added the required validation in the unit test.